src/main/java/com/laytonsmith/core/environments/GlobalEnv.java

```diff
@@ -84,14 +86,16 @@
 	 * @param root The root working directory to use
 	 * @param profiles The Profiles object to use
 	 * @param taskManager The TaskManager object to use
+	 * @param runtimeModes The {@link RuntimeMode}s for this environment.
 	 */
 	public GlobalEnv(ExecutionQueue queue, Profiler profiler, PersistenceNetwork network,
-			File root, Profiles profiles, TaskManager taskManager) {
+			File root, Profiles profiles, TaskManager taskManager, EnumSet<RuntimeMode> runtimeModes) {
```

Comment:
Nitpick that you don't need to change - this is a violation of the D in SOLID, since Set is the interface, and EnumSet is the concrete implementation, but this is fine, it will always be an EnumSet anyways.

Reply:
I do agree, but I also think that this ensures users to construct the intended efficient set here.